CodeSniffer/Standards/Generic/Sniffs/Functions/ReturnTypeDeclarationSniff.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * ReturnTypeDeclarationSniff.
+ *
+ * PHP version 7
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @copyright 2006-2014 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+/**
+ * ReturnTypeDeclarationSniff.
+ *
+ * Checks for return type declarations if the spacing between a function's
+ * closing parenthesis, colon, and return type is correct, and if
+ * the return type is lowercase.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    Arent van Korlaar <avkorlaar@hostnet.nl>
+ * @copyright 2006-2014 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ * @version   Release: @package_version@
+ * @link      http://pear.php.net/package/PHP_CodeSniffer
+ */
+class Generic_Sniffs_Functions_ReturnTypeDeclarationSniff implements PHP_CodeSniffer_Sniff
+{
+
+    /**
+     * Spacing between the function's closing parenthesis and colon.
+     *
+     * @var string
+     */
+    public $closingParenthesisColonSpacing = "";
+
+    /**
+     * Spacing between the colon and the return type.
+     *
+     * @var string
+     */
+    public $colonReturnTypeSpacing = " ";
+
+
+    /**
+     * Registers the tokens that this sniff wants to listen for.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        return array(
+                T_FUNCTION,
+                T_CLOSURE,
+               );
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+     * @param int                  $stackPtr  The position of the current token
+     * in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $nextSeparator = $this->_getClosingParenthesis($phpcsFile, $tokens, $stackPtr);
+        $endPosition   = $this->_getCharachterAfterReturnTypeDeclaration($phpcsFile, $tokens, $stackPtr);
+
+        $find = array(
+                 T_COLON,
+                 T_RETURN_TYPE,
+                 T_WHITESPACE,
+                );
+
+        $a   = $this->closingParenthesisColonSpacing;
```

Comment:
Better not to use short variable names (less than 3 symbols) to ease code reading for others.

Also the `a` doesn't really tell what's stored in it.